x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts

```diff
@@ -23,37 +30,18 @@ export function registerEnginesRoute({ router, config, log }: IRouteDependencies
       },
     },
     async (context, request, response) => {
-      try {
-        const enterpriseSearchUrl = config.host as string;
-        const { type, pageIndex } = request.query;
+      const { type, pageIndex } = request.query;
 
-        const params = querystring.stringify({
+      return enterpriseSearchRequestHandler.createRequest({
+        path: '/as/engines/collection',
+        params: `?${querystring.stringify({
           type,
           'page[current]': pageIndex,
           'page[size]': ENGINES_PAGE_SIZE,
-        });
-        const url = `${encodeURI(enterpriseSearchUrl)}/as/engines/collection?${params}`;
-
-        const enginesResponse = await fetch(url, {
-          headers: { Authorization: request.headers.authorization as string },
-        });
-
-        const engines = await enginesResponse.json();
-        const hasValidData =
-          Array.isArray(engines?.results) && typeof engines?.meta?.page?.total_results === 'number';
-
-        if (hasValidData) {
-          return response.ok({ body: engines });
-        } else {
-          // Either a completely incorrect Enterprise Search host URL was configured, or App Search is returning bad data
-          throw new Error(`Invalid data received from App Search: ${JSON.stringify(engines)}`);
-        }
-      } catch (e) {
-        log.error(`Cannot connect to App Search: ${e.toString()}`);
-        if (e instanceof Error) log.debug(e.stack as string);
-
-        return response.customError({ statusCode: 502, body: 'cannot-connect' });
-      }
+        })}`,
+        hasValidData: (body?: IEnginesResponse) =>
```

Comment:
Totally unrelated to this PR. Do we need to add this for every API request 🤔

Reply:
Nah, probably not. I think I originally added it to catch issues like the `/login` redirect (or anything else cropping up that I couldn't predict due to auth issues). I'd say it's probably only super important for pages where the front-end will crash with a typeerror of some kind if it receives unexpected data. 🤷

Reply:
To be honest I'd be fine if most API routes going forward didn't have this, and we added them on an as-needed basis (if we started getting error reports for bad API endpoints, for example).